### DISCUSSION TITLE
Forcing codec to H264 doesn't work on receiving side

### DISCUSSION THREAD

**Discussion by DerGenaue** [aiortc/aiortc#844] [Q&A]
Forcing a certain codec is explained in the `webcam.py` example:

```python
def force_codec(pc, sender, forced_codec):
    kind = forced_codec.split("/")[0]
    codecs = RTCRtpSender.getCapabilities(kind).codecs
    transceiver = next(t for t in pc.getTransceivers() if t.sender == sender)
    transceiver.setCodecPreferences(
        [codec for codec in codecs if codec.mimeType == forced_codec]
    )
```

This does not work correctly on the receiving side:

✅ The offering side (Browser) sends all the available encodings in the SDP string.

✅ The receiving side (aiortc) responds with only the forced codec in the SDP string (H264)

❌ The receiving side (aiortc) selects VP8 as the codec for the video even though the forcing worked correctly in the step before.

**Comment by marcus2vinicius** [comment]
Im need too
